### PR TITLE
Child device config management handling

### PR DESCRIPTION
**Pull request by albinsuresh**
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: crates/bin/c8y_configuration_poc/src/config_manager/upload.rs [hunk 1]
```diff
@@ -128,3 +414,10 @@ impl TryIntoOperationStatusMessage for UploadConfigFileStatusMessage {
         .to_smartrest()
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    #[test]
+    fn test() {}
+}
```
Comment (didier-wenzek) [inline crates/bin/c8y_configuration_poc/src/config_manager/upload.rs:423]: To be removed if you have no plan to add tests

**Review by didier-wenzek — APPROVED** [review]
Approved

File: crates/bin/c8y_configuration_poc/src/config_manager/upload.rs (hunk 1 above)
Comment (albinsuresh) [reply crates/bin/c8y_configuration_poc/src/config_manager/upload.rs:423]: Will be adding tests.

File: crates/bin/c8y_configuration_poc/src/config_manager/child_device.rs [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+#[cfg(test)]
+pub const FILE_TRANSFER_ROOT_PATH: &str = "/tmp";
+#[cfg(not(test))]
+pub const FILE_TRANSFER_ROOT_PATH: &str = "/var/tedge/file-transfer";
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ConfigOperationResponseTopic {
+    SnapshotResponse,
+    UpdateResponse,
+}
```
[the remark this answers is not in the document]
Comment (albinsuresh) [reply crates/bin/c8y_configuration_poc/src/config_manager/child_device.rs:23]: What I wanted was a list of topic filter `const`s that I can statically refer from anywhere. An enum felt like an easy way to group them all together. We're definitely not doing any "matching" or anything on these variants. So, that way, it probably was a misuse of enums. Could it have been a struct? Yeah, that's another option as well.

**Pull request merged**